src/tplr/comms.py

```diff
@@ -937,755 +376,106 @@ async def gather(
         init_version: Optional[str] = None,
-    ) -> tuple[bool, int, torch.optim.Optimizer, torch.optim.lr_scheduler._LRScheduler]:
-        """
-        Loads the latest checkpoint. No catchup or step simulation happens here.
-        Returns:
-            tuple: (success: bool, momentum: dict, checkpoint_current_window: int,
-                    optimizer: Optimizer, scheduler: LRScheduler)
-        """
-        init_version = init_version if init_version is not None else __version__
-        result = await self.get_latest_checkpoint(init_version)
-        if not result:
-            tplr.logger.info("No valid checkpoints found")
-            return False, 0, optimizer, scheduler
-
-        checkpoint_data, checkpoint_window = result
-        try:
-            # 1) Load model and optimizer state
-            model.load_state_dict(
-                {
-                    k: v.to(device)
-                    for k, v in checkpoint_data["model_state_dict"].items()
-                }
-            )
-            model.to(device)
-
-            for state in optimizer.state.values():
-                for k, v in state.items():
-                    if torch.is_tensor(v):
-                        state[k] = v.to(device)
-            optimizer.load_state_dict(checkpoint_data["optimizer_state_dict"])
-
-            scheduler.load_state_dict(checkpoint_data["scheduler_state_dict"])
-
-            checkpoint_start_window = checkpoint_data.get("start_window")
-            checkpoint_current_window = checkpoint_data.get("current_window")
-            checkpoint_sync_window = checkpoint_data.get("sync_window")
-            if checkpoint_start_window is None or checkpoint_current_window is None:
-                tplr.logger.warning(
-                    "Checkpoint missing start_window or current_window info"
-                )
-                return False, {}, 0, optimizer, scheduler
+    ) -> Tuple[bool, int, torch.optim.Optimizer, torch.optim.lr_scheduler.LRScheduler]:
+        """Load checkpoint - delegate to checkpoint manager"""
+        return await self.checkpoint_manager.load_checkpoint(
+            model, optimizer, scheduler, current_window, device, init_version
+        )
 
-            tplr.logger.info(
-                f"Checkpoint loaded. start_window={checkpoint_start_window}, "
-                f"checkpoint_current_window={checkpoint_current_window}, "
-                f"checkpoint_sync_window={checkpoint_sync_window}, "
-                f"local_current_window={current_window}"
-            )
+    # Peer management methods
+    async def is_miner_active(self, uid: int, recent_windows: int = 3) -> bool:
+        """Check if miner is active - delegate to peer manager"""
+        return await self.peer_manager.is_peer_active(uid, recent_windows)
 
-            self.last_checkpoint_data = checkpoint_data
+    def weighted_random_sample_no_replacement(
+        self, candidates: List[int], weights: List[int], k: int
+    ) -> List[int]:
```

Comment:
```suggestion
        self, candidates: list[int], weights: list[int], k: int
    ) -> list[int]:
```

Reply:
Actuallt realised I probably misplaces this in comms. It should in the validator only I think.